packages/cli/src/error-reporting.ts

```diff
@@ -66,9 +66,13 @@ export const initErrorHandling = async () => {
 				},
 			}),
 		],
-		beforeSend(event, { originalException }) {
+		async beforeSend(event, { originalException }) {
 			if (!originalException) return null;
 
+			if (originalException instanceof Promise) {
+				originalException = await originalException.catch((error) => error as Error);
+			}
```

Comment:
```suggestion
			if (originalException instanceof Promise) {
				originalException = await originalException.catch((error) => ensureError(error));
			}
```

Reply:
usually that's why I'd do as well. However in this case I didn't want non errors throwing inside `beforeSend` and creating some kind of a weird recursion.